code-challenges.test.js

```diff
@@ -40,11 +118,79 @@ const letterE = "e"
 
 // b) Create the function that makes the test pass.
 
+const fruits = (letterA, letterE) => {
+ if ( fruitArray.includes(letterA)) {
+  return fruitArray.slice([1, 3, 5]) 
+}
+else ( fruitArray.includes(letterE)) ; {
+ return fruitArray.slice([0, 2, 4])
+}
+}
+
+//--------------------------------------- QUESTION 2 NOTES -----------------------------------------------
+ // --So I got this error message below--
+
+// ● fruits › takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
+
+// expect(received).toEqual(expected) // deep equality
+
+// Expected: ["Mango", "Apricot", "Peach"]
+// Received: undefined
+
+//   50 |         const letterA = "a"
+//   51 |         const letterE = "e"
+// > 52 |         expect(fruits(letterA)).toEqual(["Mango", "Apricot", "Peach"])
+//      |                                 ^
+//   53 |         expect(fruits(letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//   54 |
+//   55 |     })
+
+//   at Object.toEqual (code-challenges.test.js:52:33)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       2 failed, 2 total
+// Snapshots:   0 total
+// Time:        0.184 s, estimated 1 s
+//------------------------------------------------------- So to my best understanding I used the commands that would specifically get the expected output but its just 'undefined'
+
+//off the bat I know the .filter would be the best choice but I'm unsure how to put it in the function. went with .includes method and passed the function as the argument. Did a 'return' command with the '.slice' method to identify the specified indexes and remove to get expected output.
+//---------------------------QUESTION 2 NOTES--------------------------------------------------------------
+
```

Comment:
I can tell you put a lot of thought and effort into this one.  Your pseudocode looks really good except the parameters should be the array and one letter (from the prompt `...takes in an array of words and a single letter`. ) Once you pass the array, you can use .filter and .includes.